x-pack/test/cases_api_integration/common/lib/validation.ts

```diff
@@ -25,3 +26,21 @@ export function validateCasesFromAlertIDResponse(
     expect(apiResCase.title).to.be(idToTitle.get(apiResCase.id));
   }
 }
+
+/**
+ * Compares two arrays to determine if they are sort of equal. This function returns true if the arrays contain the same
+ * elements but the ordering does not matter.
+ */
+export function arraysToEqual(array1?: object[], array2?: object[]) {
+  if (!array1 || !array2 || array1.length !== array2.length) {
+    return false;
+  }
+
+  for (const item of array1) {
```

Comment:
Let's optimize it like this:

```
const arrayAsSet = new Set(array1)
return array2.every((item) => arrayAsSet.has(item))
```

This will convert the complexity of the code from `O(n^2)` to `O(n)`